addon/-private/system/model/model.js

```diff
@@ -1631,11 +1498,13 @@ Model.reopenClass({
    @param {any} binding the value to which the callback's `this` should be bound
    */
   eachRelationship(callback, binding) {
-    get(this, 'relationshipsByName').forEach((relationship, name) => {
-      callback.call(binding, name, relationship);
-    });
+    get(this, '__internalModelClass').eachRelationship(callback, binding);
   },
 
+  __internalModelClass: computed(function() {
```

Comment:
As discussed offline, this is a static prop, and for reasons (including ember-inspector) we can't rely on setting this via `store.modelFor` lookup.  Personally I prefer this would be hidden (via WeakMap or such) from end users, but we aren't left with very good options at the moment for that.